solidity/contracts/bridge/Bridge.sol

```diff
@@ -307,34 +361,410 @@ contract Bridge is Ownable {
+
+        // Determine the swept deposits count. If main UTXO is NOT expected,
+        // all inputs should be deposits. If main UTXO is expected, one input
+        // should point to that main UTXO.
+        depositors = new address[](
+            !mainUtxoExpected ? inputsCount : inputsCount - 1
+        );
+        depositedAmounts = new uint256[](depositors.length);
+
+        // Initialize helper variables.
+        uint256 processedDepositsCount = 0;
+
+        // Inputs processing loop.
+        for (uint256 i = 0; i < inputsCount; i++) {
+            // Check if we are at the end of the input vector.
+            if (inputStartingIndex >= sweepTxInputVector.length) {
+                break;
+            }
+
+            (bytes32 inputTxHash, uint32 inputTxIndex, uint256 inputLength) =
+                parseTxInputAt(sweepTxInputVector, inputStartingIndex);
+
+            DepositInfo storage deposit =
+                deposits[
+                    uint256(
+                        keccak256(abi.encodePacked(inputTxHash, inputTxIndex))
+                    )
+                ];
+
+            if (deposit.revealedAt != 0) {
+                require(deposit.sweptAt == 0, "Deposit already swept");
+
+                if (processedDepositsCount == depositors.length) {
+                    // If this condition is true, that means a deposit input
+                    // took place of an expected main UTXO input.
+                    // In that case, we should ignore that input and let the
+                    // transaction fail at `require` checking whether expected
+                    // main UTXO was found.
+                    continue;
+                }
+
+                /* solhint-disable-next-line not-rely-on-time */
+                deposit.sweptAt = uint32(block.timestamp);
+
+                depositors[processedDepositsCount] = deposit.depositor;
+                depositedAmounts[processedDepositsCount] = deposit.amount;
+                inputsValue += depositedAmounts[processedDepositsCount];
+
+                processedDepositsCount++;
+            } else if (
+                mainUtxoExpected != mainUtxoFound &&
+                mainUtxo.txHash == inputTxHash
+            ) {
+                inputsValue += mainUtxo.txOutputValue;
+                mainUtxoFound = true;
+            } else {
+                revert("Unknown input type");
+            }
+
+            // Make the `inputStartingIndex` pointing to the next input by
+            // increasing it by current input's length.
+            inputStartingIndex += inputLength;
+        }
+
+        // Construction of the input processing loop guarantees that:
+        // `processedDepositsCount == depositors.length == depositedAmounts.length`
+        // is always true at this point. If so, we just use the first variable
+        // to assert the total count of swept deposit is bigger than zero.
+        require(
+            processedDepositsCount > 0,
```

Comment:
At this moment the wallet is allowed to include any number of deposits in a sweep.
If the number of deposits is very large, we will not be able to prove such a sweep (we will run out of gas).
Perhaps we could add a limit for the number of deposits in one sweep? 
Let's say we are sure we will never run out of gas if there are 10 or less deposits.
I would leave a TODO to add such a limit (after testing).
Also we should assume all the deposits will have `vault` set, which will increase gas consumption.

Reply:
> At this moment the wallet is allowed to include any number of deposits in a sweep.
If the number of deposits is very large, we will not be able to prove such a sweep (we will run out of gas).
Perhaps we could add a limit for the number of deposits in one sweep?
Let's say we are sure we will never run out of gas if there are 10 or fewer deposits.

I think it should be off-chain rule in the client. Even if we have a limit of 10 deposits in the contract, it does not stop the off-chain wallet to sweep more. The sweeping will happen on the Bitcoin side anyway.

> Also we should assume all the deposits will have vault set, which will increase gas consumption.

I think deposits directed to vaults will be handled in a separate PR.